crates/sui-types/src/storage.rs

```diff
@@ -56,9 +56,9 @@ pub enum DeleteKind {
 
 #[derive(Debug, PartialEq, Eq, Clone, Copy, Serialize, Deserialize)]
 pub enum MarkerKind {
-    /// An object was received at the given version in the transaction and is no longer able
-    /// to be received at that version in subequent transactions.
-    Received,
+    /// An object was received, deleted, or wrapped at the given version in the transaction and is
+    /// no longer able to be received at that version in subequent transactions.
+    ReceivedDeletedOrWrappedAtVersion,
```

Comment:
I don't have a strong opinion on whether to capture the received kind, but if we do, I would suggest we at least break this down into two variants: Received and OwnedTombstone, such that this can be a natural replacement for tombstones if we need to, instead of binding to the need of transfer to object.

Reply:
I like it! This will also make the logic nicer around signing, TM, and object loading -- we use the received markers for signing and when loading, but the `OwnedTombstones` for TM checks. I'll work on making this change today.